### DISCUSSION TITLE
Raspberry Pi 4 Model B support, VS Code plugin

### DISCUSSION THREAD

**Discussion by deboboy** [microsoft/devicescript#413] [Q&A] [answered]
Hola,
Any plans to support Raspberry Pi 4 B?  Didn't see it explicitly listed in the documentation.

Also, when will a VS Code plugin be available?

**Comment by pelikhan** [comment, marked as answer]
Hola,

The VSCode extension information is at https://microsoft.github.io/devicescript/getting-started/vscode

As for the Pi 4 B, are you referring to running vcode and developer devicescript on it? It should just work but unfortunately we don't have a pi 4 to test this.

**Reply by deboboy** [reply]
Si!

I have an RPi4B that’s ready for this experiment.  Install then report any issues?

**Reply by pelikhan** [reply]
It Should Work (TM). We develop on macos, windows and codespaces so as long as node.js and serialport is supported, things should work.https://www.npmjs.com/package/serialport

**Reply by deboboy** [reply]
Update:
Installed the VS Code extension on my Pi 4 B.  The Start Simulator or Connect Device buttons don't do anything.  Assume a service would have started [with output in a VS Code panel].

Is there a log somewhere I can inspect?

**Reply by pelikhan** [reply]
Could you try the cli (https://microsoft.github.io/devicescript/api/cli)? The VScode extension launches it in the background and that is what is probably failing.

I remember that the pi may have some old version of node and we require node LTS (18).

**Reply by pelikhan** [reply]
Here are instruction to force devicescript to use a shell. That should give you the ability to look at why the command line tool failed.

https://microsoft.github.io/devicescript/getting-started/vscode/troubleshooting#devicescript-terminal-wont-start

**Reply by deboboy** [reply]
Having problems getting the CLI install to work...
gyp ERR! stack Error: Python executable "/usr/bin/python" is v3.9.2, which is not supported by gyp.

Will try to align python libs; see if that gets the CLI to install

**Reply by pelikhan** [reply]
I would suspect that there is prebuild for "serialport", the npm package that does serial comma. This, gyp is called and it does not find python.

What version of node.js do you run?

**Reply by deboboy** [reply]
v8.10.0

**Reply by pelikhan** [reply]
Oh you'll need to update to v18

**Reply by deboboy** [reply]
That's odd because I selected the current build from the Node repo [for this device].  Will try to find v18.

**Reply by pelikhan** [reply]
maybe you're getting the builtin outdated node install?

Does "nvm" work on raspi?

**Reply by pelikhan** [reply]
I'm adding a check to test which version of node devicescript will try to run. https://github.com/microsoft/devicescript/pull/493

Note that there is a setting where you can specify the path to the node executable (DeviceScript user settings)

**Comment by pelikhan** [comment]
@deboboy appday is 10 year old yesterday.

**Reply by deboboy** [reply]
Wow, I still have a t-shirt
